snuba/web/rpc/v1/resolvers/R_eap_spans/common/sampling_in_storage_util.py

```diff
@@ -286,12 +292,22 @@ def _emit_estimation_error_info(
 
 @with_span(op="function")
 def run_query_to_correct_tier(
-    in_msg: T,
+    in_msg: TraceItemTableRequest | TimeSeriesRequest,
     query_settings: HTTPQuerySettings,
     timer: Timer,
-    build_query: Callable[[T], Query],
+    build_query: Callable[[TraceItemTableRequest | TimeSeriesRequest], Query],
```

Comment:
pull out the build_query code into its own file for each resolver, import that in the routing_strategy base

Reply:
or just pass in the functions and ignore the type